ros2bag/ros2bag/verb/record.py

```diff
@@ -211,83 +211,63 @@ def _check_necessary_argument(self, args):
         # At least one options out of --all, --all-topics, --all-services, --services, --topics,
         # --topic-types or --regex must be used
         if not (args.all or args.all_topics or args.all_services or
-           args.services or (args.topics and len(args.topics) > 0) or
-           (args.topic_types and len(args.topic_types) > 0) or args.regex):
+                args.services or (args.topics and len(args.topics) > 0) or
```

Comment:
It is better to change the check of args.services like args.topics.   

```suggestion
                (args.services and len(args.services)>0) or (args.topics and len(args.topics) > 0) or
```

Reply:
Fixed.